lib/backend/k8s/k8s_test.go

```diff
@@ -1967,6 +1987,154 @@ var _ = testutils.E2eDatastoreDescribe("Test Watch support", testutils.Datastore
 		ctx = context.Background()
 	})
 
+	It("Should support watching Profiles", func() {
+		By("Watching a single Profile", func() {
+			name := "default"
+			watch, err := c.Watch(ctx, model.ResourceListOptions{Name: name, Kind: apiv3.KindProfile}, "")
+			Expect(err).NotTo(HaveOccurred())
+			event := ExpectAddedEvent(watch.ResultChan())
+			Expect(event.New.Key.String()).To(Equal("Profile(kns.default)"))
+		})
+		By("Watching all Profiles", func() {
+			watch, err := c.Watch(ctx, model.ResourceListOptions{Kind: apiv3.KindProfile}, "")
+			Expect(err).NotTo(HaveOccurred())
+			ExpectAddedEvent(watch.ResultChan())
+		})
+	})
+
+	It("Should support watching NetworkPolicies", func() {
+		createTestNetworkPolicy := func() {
+			np := networkingv1.NetworkPolicy{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-watch-net-policy",
+				},
+			}
+			res := c.ClientSet.NetworkingV1().RESTClient().Post().Resource("networkpolicies").Namespace("default").Body(&np).Do()
+			Expect(res.Error()).NotTo(HaveOccurred())
+		}
+		deleteAllNetworkPolicies := func() {
```

Comment:
Ginkgo supports an `AfterEach()` option which might be well-suited for this. It ensures you run this after each test, even if the test itself fails.

The way it is currently written, an `Expect` might fail and then this function won't be executed.

Reply:
Ok, I'm redoing this with `Describe` and `It` (instead of `By`) so I can use `Before/AfterEach`